### PR TITLE
Fix 2774

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

**Comment by antgonza** [comment]
Forgot screenshot: 
checksum_size

**Comment by antgonza** [comment]
Some highlights of this PR:
- fixed #2774, which discovered a few things and the reason there are more fixes here
- retrieve_filepaths used to return a list now returns a dict; this change was cause originally I kept returning a list and realized that it was crazy to keep track of what element was what, see: https://github.com/biocore/qiita/pull/2833
- had to fix all tests so the previous changed worked
- some tests were using artifact 1 as the initial artifact for tests, which in some cases deleted or altered its files; decided to always create a new artifact for these cases so the files are not changed, so the checksum/size do not change
- purge_files_from_filesystem was one of these tests/methods and also found a bug on how the legacy files were being dealt with; which is something we saw the first time we ran it on main qiita
- compute_checksum had a bug which was only present in python2 (which is how all the checksums are currently calculated) so decided to rewrite it to be python2/3 safe and recalculate all the checksums during the patch

File: qiita_db/environment_manager.py [hunk 1]
```diff
@@ -436,5 +436,6 @@ def patch(patches_dir=PATCHES_DIR, verbose=False, test=False):
         # for the test Study (1) so alot of the tests actually expect this.
```
Comment (charles-cowart) [inline qiita_db/environment_manager.py]: Outside the scope of this change, but if you can change 'alot' -> 'a lot', that would be awesome ty.

File: qiita_db/util.py [hunk 2]
```diff
@@ -422,17 +420,17 @@ def compute_checksum(path):
...
     for fp in filepaths:
-        with open(fp, 'rb', newline=None) as f:
-            # Go line by line so we don't need to load the entire file
-            for line in f:
-                if crc is None:
-                    crc = crc32(line)
-                else:
-                    crc = crc32(line, crc)
-    # We need the & 0xffffffff in order to get the same numeric value across
+        with open(fp, 'rb') as f:
```
Comment (charles-cowart) [inline qiita_db/util.py]: Just to confirm, removing 'newline=None' from the open call was intended?

File: qiita_pet/handlers/artifact_handlers/base_handlers.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 from os.path import basename, relpath
 from json import dumps
+from humanize import naturalsize
```
Comment (charles-cowart) [inline qiita_pet/handlers/artifact_handlers/base_handlers.py]: Are you sure it's all right to use humanize? I looks like it last had a release just before 2015:

https://pypi.org/project/humanize/

File: qiita_db/environment_manager.py (hunk 1 above)
Comment (antgonza) [reply qiita_db/environment_manager.py]: k

File: qiita_db/util.py (hunk 2 above)
Comment (antgonza) [reply qiita_db/util.py]: Yup ... it's automatic.

File: qiita_pet/handlers/artifact_handlers/base_handlers.py (hunk 3 above)
Comment (antgonza) [reply qiita_pet/handlers/artifact_handlers/base_handlers.py]: Yup, we have been using it for a while; note that it has been part of the setup.py for a while ...

**Comment by antgonza** [comment]
Thanks!!

**Comment by charles-cowart** [comment]
no problem! Sorry for the delay

**Pull request merged**